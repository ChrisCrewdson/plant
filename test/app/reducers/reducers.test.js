// @ts-ignore - static hasn't been defined on seamless types yet.
const seamless = require('seamless-immutable').static;
const rootReducer = require('../../../app/reducers');
const actions = require('../../../app/actions');


describe('/app/reducers', () => {
  test('should reduce a logout success', () => {
    const actual = rootReducer(seamless({}), actions.logoutSuccess());
    expect(actual).toMatchSnapshot();
  });

  test('should reduce a logout request', () => {
    const actual = rootReducer(seamless({}), actions.logoutRequest());
    expect(actual).toMatchSnapshot();
  });

  test('should reduce a logout failure', () => {
    const actual = rootReducer(seamless({}), actions.logoutFailure());
    expect(actual).toMatchSnapshot();
  });

  describe('sanity check all reducers', () => {
    ['interim', 'locations', 'notes', 'plants', 'user', 'users']
      .forEach((reducer) => {
        test(`${reducer} should not have an undefined reducer`, () => {
          // eslint-disable-next-line
          const { reducers } = require(`../../../app/reducers/${reducer}`);
          expect(reducers).toBeInstanceOf(Object);
          expect(reducers.undefined).toBeUndefined();
          Object.keys(reducers).forEach((reducerKey) => {
            const value = reducers[reducerKey];
            expect(value).toBeInstanceOf(Function);
          });
        });
      });
  });
});
