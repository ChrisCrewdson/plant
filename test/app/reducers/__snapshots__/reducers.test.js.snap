// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/app/reducers should reduce a logout failure 1`] = `
Object {
  "interim": Object {},
  "locations": Object {},
  "notes": Object {},
  "plants": Object {},
  "user": Object {
    "isLoggedIn": false,
    "status": "failed",
  },
  "users": Object {},
}
`;

exports[`/app/reducers should reduce a logout request 1`] = `
Object {
  "interim": Object {},
  "locations": Object {},
  "notes": Object {},
  "plants": Object {},
  "user": Object {
    "isLoggedIn": false,
    "status": "logout",
  },
  "users": Object {},
}
`;

exports[`/app/reducers should reduce a logout success 1`] = `
Object {
  "interim": Object {},
  "locations": Object {},
  "notes": Object {},
  "plants": Object {},
  "user": Object {
    "isLoggedIn": false,
    "status": "logout",
  },
  "users": Object {},
}
`;
